lib/ui/views/homepage/widgets/product_item.dart

```diff
@@ -0,0 +1,111 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:receipe_app/data_model/product_model.dart';
+import 'package:receipe_app/ui/common/app_images.dart';
+import 'package:receipe_app/ui/extension/app_typography.dart';
+import 'package:receipe_app/ui/extension/palette.dart';
+
+class ProductItem extends StatelessWidget {
+  final ProductModel productModel;
+  const ProductItem({
+    super.key,
+    required this.productModel,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    ThemeData theme = Theme.of(context);
+    AppTypography? typography = theme.extension<AppTypography>();
+    Palette? palette = theme.extension<Palette>();
+    return Card(
+      elevation: 0,
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.start,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Stack(
+            alignment: AlignmentDirectional.topEnd,
+            children: [
+              Image.asset(
+                productModel.image,
+                height: 200.h,
+                width: double.infinity,
+              ),
+              productModel.isLiked
+                  ? Padding(
+                      padding: EdgeInsets.all(12.0.h),
+                      child: SvgPicture.asset(AppImages.unLikedIcon),
+                    )
+                  : Padding(
+                      padding: EdgeInsets.all(12.0.h),
+                      child: SvgPicture.asset(AppImages.lkedIcon),
+                    )
```

Comment:
use a builder widget rather than a ternary operator so code will be clean. eg
```
Builder(
 builder: (context){
  if (productModel.isLiked){
    return Padding(
                      padding: EdgeInsets.all(12.0.h),
                      child: SvgPicture.asset(Apmages.unLikedIcon),
                    );
  
} 
 return Padding(
                      padding: EdgeInsets.all(12.0.h),
                      child: SvgPicture.asset(AppImages.lkedIcon),
                    );
}
)
```

Reply:
closed